service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/database/DeleteAzureStorageContainerStep.java

```diff
@@ -0,0 +1,38 @@
+package bio.terra.workspace.service.resource.controlled.flight.clone.azure.database;
+
+import bio.terra.stairway.*;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.service.resource.controlled.ControlledResourceService;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DeleteAzureStorageContainerStep implements Step {
+  private static final Logger logger =
+      LoggerFactory.getLogger(DeleteAzureStorageContainerStep.class);
+  private final ControlledResourceService controlledResourceService;
+  private final String storageContainerName;
+  private final UUID resourceId;
+
+  public DeleteAzureStorageContainerStep(
+      String storageContainerName,
+      UUID resourceId,
+      ControlledResourceService controlledResourceService) {
+    this.controlledResourceService = controlledResourceService;
+    this.storageContainerName = storageContainerName;
+    this.resourceId = resourceId;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    return new CreateAzureStorageContainerStep(
+            storageContainerName, resourceId, controlledResourceService)
+        .undoStep(context);
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext context) throws InterruptedException {
+    logger.error("Cannot undo deletion of Azure Storage Container resource {}.", resourceId);
```

Comment:
TOL: Should this be a warning instead of error? Or possibly nothing since there isn't anything to undo (similar to launch flights undo step)?

Reply:
I took this from the storage container's delete step - I don't think it matters too much since the end result is still a step "success", but it would be pretty weird if we were undoing this step since it should be the final step besides setting the response body.
 https://github.com/DataBiosphere/terra-workspace-manager/blob/aaddf30660d712079714356daca675c8f7a3e81c/service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storageContainer/DeleteAzureStorageContainerStep.java#L130-L134